### PR TITLE
Pytorch example with alluxiofs

### PR DESCRIPTION
**Pull request by SibylYang**
An example that uses Alluxiofs to speed up pytorch distributed NLP training with large CSV files.
Formatted the file using black and reorder_python_imports.

### CODE REVIEW

**Comment by SibylYang** [comment]
@lucyge2022 @LuQQiu For your review. Thanks!

File: examples/pytorch_distributed_BERT_training.py [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+    Preprocess each file in the directory for Dataset class.
+    :param directory_path: directory path contain the files to be processed
+    :param chunk_size: read file that has chunk_size to the memory each time
+    :return: processed_file_info is a dictionary that contain start_line number for each file;
+            total length of all files in the directory
+    """
+    fsspec.register_implementation("alluxiofs", AlluxioFileSystem, clobber=True)
+    alluxio_fs = fsspec.filesystem(
+        "alluxiofs", etcd_hosts="localhost", etcd_port=2379, target_protocol="s3"
+    )
```
Comment (lucyge2022) [inline examples/pytorch_distributed_BERT_training.py:202]: so in here, we are already in a child process right?
we can reuse the alluxio_fs instance created per child process when we spawn main() correct?

**Comment by SibylYang** [comment]
@lucyge2022 Moved alluxio_fs initialization and file preprocessing to the parent process. Add dependencies in the top comment lines.

File: examples/pytorch_distributed_BERT_training.py [hunk 2]
```diff
@@ -0,0 +1,356 @@
...
+        chunk_iterator = pd.read_csv(
+            self.alluxio_fs.open(target_file_name, mode="r"),
+            chunksize=self.read_chunk_size,
+        )
+
+        for i, chunk in enumerate(chunk_iterator):
+            # only the chunk will be loaded into memory each time
+            # memory occupied by chunk will be freed once outside the loop
+            if i == chunk_number:
+                target_line = chunk.iloc[line_within_chunk]
```
Comment (lucyge2022) [inline examples/pytorch_distributed_BERT_training.py:99]: break here if matched line is found

File: examples/pytorch_distributed_BERT_training.py [hunk 3]
```diff
@@ -0,0 +1,356 @@
...
+        ids = inputs["input_ids"]
+        token_type_ids = inputs["token_type_ids"]
+        mask = inputs["attention_mask"]
+
+        ## record behavior to a output file
+        # self.total_access += 1
+        # with open(self.output_filename, 'a') as file:
+        #     file.write(
+        #         f'access to global index {index}, which is line {target_line_index} in file {target_file_name}: {target_line.iloc[0]}\n')
+        #     file.write(f'__getitem__ total access: {self.total_access}\n')
```
Comment (lucyge2022) [inline examples/pytorch_distributed_BERT_training.py:120]: remove if not needed

**Review by lucyge2022 — APPROVED** [review]
LGTM besides comments

**Comment by lucyge2022** [comment]
@SibylYang  also there's some test case failure, the on in CI/lint seems to be bcos of style, can u use pre-commit tool (apt-get install pre-commit) to screen thru the scripts, it will correct the code style for you.
for the CI/Tests , can u make this change to file tests/conftest.py

**Comment by SibylYang** [comment]
@lucyge2022 code updated to address your comments

**Review by lucyge2022 — APPROVED** [review]

**Pull request merged**